## Visión general

In [66]:
import pandas as pd

df = pd.read_csv("https://media.githubusercontent.com/media/elkofix/cardiac-risk-db/refs/heads/master/completa.csv")

In [67]:
df.shape

(2782908, 29)

In [68]:
df.head()

11_CodEtnia_Indigena  11_CodEtnia_Negro  11_CodEtnia_Otro  \
0                   0.0                0.0               1.0   
1                   0.0                0.0               1.0   
2                   0.0                0.0               1.0   
3                   0.0                0.0               1.0   
4                   0.0                0.0               1.0   

   11_CodEtnia_Palenquero de San Basilio  11_CodEtnia_ROM (Gitano)  \
0                                    0.0                       0.0   
1                                    0.0                       0.0   
2                                    0.0                       0.0   
3                                    0.0                       0.0   
4                                    0.0                       0.0   

   11_CodEtnia_Raizal  ENDOSALUD_SI  \
0                 0.0           0.0   
1                 0.0           1.0   
2                 0.0           0.0   
3                 0.0           0.0   
4                 0.0           0.0   

   Indice Masa Corporal = Peso/talla Al Cuadrado  24_Talla  23_Peso  \
0                                      26.892323     167.0     75.0   
1                                      31.250000     160.0     80.0   
2                                      24.690405     161.0     64.0   
3                                      25.000000     160.0     64.0   
4                                      23.437500     160.0     60.0   

   25_TenArtSis  26_TenArtDitlica  36_RcbeIECA_SI  37_RcbeARA2_SI  \
0         130.0              89.0             0.0             0.0   
1         130.0              70.0             0.0             0.0   
2         160.0              80.0             0.0             0.0   
3         160.0              80.0             0.0             0.0   
4         110.0              80.0             0.0             0.0   

   22_EtiologiaERC_HTA-DM  EDAD (Años cumplidos)  8_Sexo_F  \
0                       0                   81.0         0   
1                       0                   55.0         0   
2                       0                   65.0         1   
3                       1                   78.0         1   
4                       0                   55.0         1   

   9_Regimen_CONTRIBUTIVO  9_Regimen_NO ASEGURADO  9_Regimen_SUBSIDIADO  \
0                       0                       0                     1   
1                       0                       0                     1   
2                       0                       0                     1   
3                       0                       1                     0   
4                       0                       0                     1   

   18_DxHTA_SI  18_DxHTA_NO  22_EtiologiaERC_HTA  22_EtiologiaERC_DM  \
0            1            0                    1                   0   
1            1            0                    1                   0   
2            1            0                    1                   0   
3            1            0                    0                   0   
4            0            1                    0                   1   

   CLASIFICACION FINAL DEL RIESGO_ALTO  CLASIFICACION FINAL DEL RIESGO_BAJO  \
0                                    0                                    1   
1                                    0                                    0   
2                                    0                                    0   
3                                    1                                    0   
4                                    1                                    0   

   CLASIFICACION FINAL DEL RIESGO_MODERADO  20_DxDM_SI  \
0                                        0           0   
1                                        1           0   
2                                        1           0   
3                                        0           1   
4                                        0           1   

   CLASIFICACION FINAL DEL RIESGO_MUY A

In [69]:
df.dtypes

11_CodEtnia_Indigena                             float64
11_CodEtnia_Negro                                float64
11_CodEtnia_Otro                                 float64
11_CodEtnia_Palenquero de San Basilio            float64
11_CodEtnia_ROM (Gitano)                         float64
11_CodEtnia_Raizal                               float64
ENDOSALUD_SI                                     float64
Indice Masa Corporal = Peso/talla Al Cuadrado    float64
24_Talla                                         float64
23_Peso                                          float64
25_TenArtSis                                     float64
26_TenArtDitlica                                 float64
36_RcbeIECA_SI                                   float64
37_RcbeARA2_SI                                   float64
22_EtiologiaERC_HTA-DM                             int64
EDAD (Años cumplidos)                            float64
8_Sexo_F                                           int64
9_Regimen_CONTRIBUTIVO                             int64
9_Regimen_NO ASEGURADO                             int64
9_Regimen_SUBSIDIADO                               int64
18_DxHTA_SI                                        int64
18_DxHTA_NO                                        int64
22_EtiologiaERC_HTA                                int64
22_EtiologiaERC_DM                                 int64
CLASIFICACION FINAL DEL RIESGO_ALTO                int64
CLASIFICACION FINAL DEL RIESGO_BAJO                int64
CLASIFICACION FINAL DEL RIESGO_MODERADO            int64
20_DxDM_SI                                         int64
CLASIFICACION FINAL DEL RIESGO_MUY ALTO            int64
dtype: object

En este caso identifico que hay muchas variables categóricas que estan almancenadas como int64 y float64, es significa que para cada uno de los valores estoy usando 8 bytes, sin embargo solo necesito representar unos y ceros, por lo que una opción es pasar las variables binarias a booleanos, que solo ocupan 1 byte por valor.

In [71]:
bin_cols = [col for col in df.columns if set(df[col].unique()) <= {0, 1}]
df[bin_cols] = df[bin_cols].astype('bool')

In [72]:
df.dtypes

11_CodEtnia_Indigena                                bool
11_CodEtnia_Negro                                   bool
11_CodEtnia_Otro                                    bool
11_CodEtnia_Palenquero de San Basilio               bool
11_CodEtnia_ROM (Gitano)                            bool
11_CodEtnia_Raizal                                  bool
ENDOSALUD_SI                                        bool
Indice Masa Corporal = Peso/talla Al Cuadrado    float64
24_Talla                                         float64
23_Peso                                          float64
25_TenArtSis                                     float64
26_TenArtDitlica                                 float64
36_RcbeIECA_SI                                      bool
37_RcbeARA2_SI                                      bool
22_EtiologiaERC_HTA-DM                              bool
EDAD (Años cumplidos)                            float64
8_Sexo_F                                            bool
9_Regimen_CONTRIBUTIVO                              bool
9_Regimen_NO ASEGURADO                              bool
9_Regimen_SUBSIDIADO                                bool
18_DxHTA_SI                                         bool
18_DxHTA_NO                                         bool
22_EtiologiaERC_HTA                                 bool
22_EtiologiaERC_DM                                  bool
CLASIFICACION FINAL DEL RIESGO_ALTO                 bool
CLASIFICACION FINAL DEL RIESGO_BAJO                 bool
CLASIFICACION FINAL DEL RIESGO_MODERADO             bool
20_DxDM_SI                                          bool
CLASIFICACION FINAL DEL RIESGO_MUY ALTO             bool
dtype: object

In [73]:
import pandas as pd

numericas = df.select_dtypes(include=['float64'])
resumen_numericas = numericas.describe().T

categoricas = df.select_dtypes(include=['bool'])
valores_unicos = categoricas.nunique()

print("Resumen de variables numéricas")
display(resumen_numericas)
print("\n Valores únicos en categóricas/bool")
display(valores_unicos)


Resumen de variables numéricas


count        mean  \
Indice Masa Corporal = Peso/talla Al Cuadrado  2782908.0   28.497291   
24_Talla                                       2782908.0  161.994512   
23_Peso                                        2782908.0   74.417583   
25_TenArtSis                                   2782908.0  125.383537   
26_TenArtDitlica                               2782908.0   77.872443   
EDAD (Años cumplidos)                          2782908.0   63.378973   

                                                     std   min         25%  \
Indice Masa Corporal = Peso/talla Al Cuadrado  21.139211   0.0   24.609734   
24_Talla                                        8.483833  10.0  157.000000   
23_Peso                                        15.463369   0.0   64.000000   
25_TenArtSis                                   15.510102   0.0  120.000000   
26_TenArtDitlica                                9.051994   0.0   70.000000   
EDAD (Años cumplidos)                          14.567678   6.0   54.000000   

                                                      50%     75%     max  
Indice Masa Corporal = Peso/talla Al Cuadrado   27.688778   31.25  6500.0  
24_Talla                                       162.000000  168.00   260.0  
23_Peso                                         73.000000   83.00   258.0  
25_TenArtSis                                   120.000000  130.00   270.0  
26_TenArtDitlica                                80.000000   80.00   191.0  
EDAD (Años cumplidos)                           64.000000   74.00   115.0


 Valores únicos en categóricas/bool


11_CodEtnia_Indigena                       2
11_CodEtnia_Negro                          2
11_CodEtnia_Otro                           2
11_CodEtnia_Palenquero de San Basilio      2
11_CodEtnia_ROM (Gitano)                   2
11_CodEtnia_Raizal                         2
ENDOSALUD_SI                               2
36_RcbeIECA_SI                             2
37_RcbeARA2_SI                             2
22_EtiologiaERC_HTA-DM                     2
8_Sexo_F                                   2
9_Regimen_CONTRIBUTIVO                     2
9_Regimen_NO ASEGURADO                     2
9_Regimen_SUBSIDIADO                       2
18_DxHTA_SI                                2
18_DxHTA_NO                                2
22_EtiologiaERC_HTA                        2
22_EtiologiaERC_DM                         2
CLASIFICACION FINAL DEL RIESGO_ALTO        2
CLASIFICACION FINAL DEL RIESGO_BAJO        2
CLASIFICACION FINAL DEL RIESGO_MODERADO    2
20_DxDM_SI                                 2
CLASIFICACION FINAL DEL RIESGO_MUY ALTO    2
dtype: int64

Al revisar las variables principales veo que hay varios errores o valores extremos que claramente no son realistas:  

**IMC (Indice de Masa Corporal):** la media está alrededor de 28, lo que tiene sentido para una población con sobrepeso leve. Sin embargo, el mínimo es 0 y el máximo 6500, lo cual es imposible. Probablemente hay registros mal calculados, quizá la talla estaba en cm y no se convirtió a metros. Valores realistas deberían estar entre 10 y 60.  

**Talla (cm):** la media es ~162 cm, bastante lógica. Pero hay valores mínimos de 10 cm y máximos de 260 cm. 10 cm es claramente un error y 260 cm sería un caso extraordinario (la persona más grande en colombia mide 235 cm).

**Peso (kg):** la media es ~74 kg, coherente. Sin embargo, hay registros de 0 kg (error) y hasta 258 kg, que aunque posible, es extremadamente raro. La mayoría de los casos deberían estar entre 30 y 200 kg.  

**Tensión arterial sistólica (mmHg):** media de ~125 mmHg, lógica. Min=0 (imposible) y máx=270 mmHg, extremadamente alto. Valores normales suelen estar entre 90 y 200 mmHg.  

**Tensión arterial diastólica (mmHg):** media ~78 mmHg, normal. Min=0 y máx=191 mmHg, que es casi incompatible con la vida, claramente valores atípicos.  

**Edad (años):** media ~63 años, coherente. Mínimo de 6 años sugiere que hay niños incluidos y máximo de 115 años, posible pero muy poco frecuente.  

En general, estos datos necesitan limpieza: muchos ceros, máximos imposibles y outliers que pueden distorsionar análisis posteriores.
